editor/src/components/canvas/canvas-strategies/strategies/reparent-utils.ts

```diff
@@ -162,3 +174,44 @@ export function cursorForMissingReparentedItems(
 
   return null
 }
+
+export function placeholderCloneCommands(
+  canvasState: InteractionCanvasState,
+  customStrategyState: CustomStrategyState,
+  filteredSelectedElements: Array<ElementPath>,
+  newParent: ElementPath,
+): { commands: Array<CanvasCommand>; duplicatedElementNewUids: { [elementPath: string]: string } } {
+  let duplicatedElementNewUids: { [elementPath: string]: string } = {
+    ...customStrategyState.duplicatedElementNewUids,
+  }
+  const commands = filteredSelectedElements.flatMap((elementPath) => {
+    const element = MetadataUtils.findElementByElementPath(
+      canvasState.startingMetadata,
+      elementPath,
+    )
+    // we want to keep a placeholder element where the original dragged element was to avoid the new parent shifting around on the screen
+    const hasCommonAncestor =
+      EP.getCommonParent([newParent, EP.parentPath(elementPath)]) != null &&
+      !MetadataUtils.isPositionAbsolute(element)
+    if (hasCommonAncestor) {
+      const selectedElementString = EP.toString(elementPath)
+      const newUid =
+        customStrategyState.duplicatedElementNewUids[selectedElementString] ??
+        generateUidWithExistingComponents(canvasState.projectContents)
+      duplicatedElementNewUids[selectedElementString] = newUid
+
+      const newPath = EP.appendToPath(EP.parentPath(elementPath), newUid)
+
+      return [
+        duplicateElement('mid-interaction', elementPath, newUid),
+        wildcardPatch('mid-interaction', {
+          hiddenInstances: { $push: [newPath] },
+        }),
+        hideInNavigatorCommand([newPath]),
```

Comment:
So I guess we should use something like this for other re-parenting strategies (but using the old path in those)?